xbmc/Application.cpp

```diff
@@ -2939,6 +2939,42 @@ void CApplication::Stop(int exitCode)
   Sleep(200);
 }
 
+bool CApplication::ResumePlayer(const CFileItemPtr& itemIn) {
+  if (itemIn->IsEmpty())
+  {
+    CLog::Log(LOGDEBUG, "CApplication::ResumePlayer nothing to resume");
+    return false;
+  }
+
+  CLog::Log(LOGDEBUG, "CApplication::ResumePlayer resume item %s", itemIn->GetPath().c_str());
+  if (m_pPlayer->HasPlayer())
+  {
+    CLog::Log(LOGWARNING, "CApplication::ResumePlayer can't resume, player is already startet");
+    return false;
+  }
+
+  CFileItem item(itemIn->GetItemToPlay());
+  CLog::Log(LOGDEBUG, "CApplication::ResumePlayer resume item to play %s", item.GetPath().c_str());
+#ifdef HAS_DVD_DRIVE
+  if (item.IsDVD() || item.IsCDDA())
+    return MEDIA_DETECT::CAutorun::PlayDisc(item.GetPath(), true, false);
+#endif
+
+  item.m_lStartOffset = STARTOFFSET_RESUME;
+
+  if (item.IsVideoDb())
+  {
+    item.SetProperty("original_listitem_url", item.GetPath());
+    item.SetPath(item.GetVideoInfoTag()->m_strFileNameAndPath);
+  }
+  item.SetProperty("check_resume", false);
+
+  if (item.IsLiveTV()) // pvr tv or pvr radio?
+    return g_application.PlayMedia(item, "", PLAYLIST_NONE);
+  else
+    return CServiceBroker::GetPlaylistPlayer().Play(std::make_shared<CFileItem>(item), "");
```

Comment:
This could be removed, g_application.PlayMedia does already handle isPlaylist and isPVR very well